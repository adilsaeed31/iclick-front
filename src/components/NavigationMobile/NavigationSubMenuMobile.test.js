import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "mobx-react";
import NavigationSubMenuMobile from "./NavigationSubMenuMobile";

const mockNavItemWithSubItems = {
  items: [{
    items: [{
      items: [],
      navigationItem: {
        data: {
          classNames: null,
          content: [{
            language: "en",
            value: "NavSubSubItem"
          }],
          isUrlRelative: true,
          shouldOpenInNewWindow: false,
          url: "/nav-sub-sub-item"
        }
      }
    }],
    navigationItem: {
      data: {
        classNames: null,
        content: [{
          language: "en",
          value: "NavSubItem"
        }],
        isUrlRelative: true,
        shouldOpenInNewWindow: false,
        url: "/nav-sub-item"
      }
    }
  }],
  navigationItem: {
    data: {
      classNames: null,
      content: [{
        language: "en",
        value: "NavItem"
      }],
      isUrlRelative: true,
      shouldOpenInNewWindow: false,
      url: "/nav-item"
    }
  }
};


const uiStore = {
  closeMenuDrawer() {},
  toggleMenuDrawerOpen() {}
};

const routingStore = {
  tagId: "v2hE7fzL6cJniwgSm"
};

test("basic snapshot", () => {
  const component = renderer.create((
    <Provider primaryShopId={"J8Bhq3uTtdgwZx3rz"} routingStore={routingStore} uiStore={uiStore}>
      <NavigationSubMenuMobile navItem={mockNavItemWithSubItems} />
    </Provider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
