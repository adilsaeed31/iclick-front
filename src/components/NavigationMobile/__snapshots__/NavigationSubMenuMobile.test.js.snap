// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<div
  className="SkNavigationItemMobile-root-1"
>
  <div
    className="SkNavigationItemMobile-header-2"
  >
    <div
      className="MuiToolbar-root-5 MuiToolbar-regular-7"
    >
      <h6
        className="MuiTypography-root-9 MuiTypography-subtitle1-27 SkNavigationItemMobile-toolbarTitle-3"
      />
      <button
        className="MuiButtonBase-root-51 MuiIconButton-root-45"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-50"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-54"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-101"
        />
      </button>
    </div>
    <hr
      className="MuiDivider-root-63"
    />
  </div>
  <div
    className="SkNavigationItemMobile-menu-4"
  >
    <div
      className="MuiList-root-68"
      onBlur={[Function]}
      onKeyDown={[Function]}
      role="menu"
    >
      <div
        className="MuiButtonBase-root-51 MuiListItem-root-78 MuiListItem-default-81 MuiListItem-gutters-86 MuiListItem-button-87"
        color="inherit"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <div
          className="MuiListItemText-root-90"
        />
        <div
          className="MuiListItemIcon-root-96"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-54"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
            />
          </svg>
        </div>
        <span
          className="MuiTouchRipple-root-101"
        />
      </div>
      <hr
        className="MuiDivider-root-63"
      />
    </div>
  </div>
</div>
`;
